src/ui/App.tsx

```diff
@@ -55,16 +63,39 @@ function App(): JSX.Element {
 
   const [loadingThreads, setLoadingThreads] = useState(LoadState.initial);
 
+  async function* fetchThreadsWithMetadata(): AsyncGenerator<Thread> {
+    // TODO: Sort by date.
```

Comment:
Sorting by date requires having fetch messages unfortunately, so we won't be able to do this until we are fully syncing messages ahead of time and we make this function not do that work incrementally anymore.